rp2040-hal/src/usb.rs

```diff
@@ -498,14 +509,22 @@ impl UsbBusTrait for UsbBus {
             // TODO: check for suspend request
             // TODO: check for resume request
 
-            // check for bus reset
-            let sie_status = inner.ctrl_reg.sie_status.read();
-            if sie_status.bus_reset().bit_is_set() {
-                return PollResult::Reset;
-            }
-
+            let sie_status_flags = inner.ctrl_reg.sie_status.read();
             let (mut ep_out, mut ep_in_complete, mut ep_setup): (u16, u16, u16) = (0, 0, 0);
 
+            // check for setup request
+            // Only report setup if OUT has been cleared.
+            if sie_status_flags.setup_rec().bit_is_set() {
+                // Clear the setup_rec interrupt flag
+                inner
+                    .ctrl_reg
+                    .sie_status
+                    .modify(|_, w| w.setup_rec().set_bit());
```

Comment:
This bit is cleared line [286](https://github.com/rp-rs/rp-hal/blob/af862424633107515fc3ac5f2f044474999f7511/rp2040-hal/src/usb.rs#L286).
Also it must not be cleared until it has been treated (see [here](https://docs.rs/usb-device/0.2.8/usb_device/bus/enum.PollResult.html#variant.Data.field.ep_setup))

Reply:
I'll revert this section to how it was previously